lib/web/files.go

```diff
@@ -148,6 +149,9 @@ func (h *Handler) transferFile(w http.ResponseWriter, r *http.Request, p httprou
 
 	err = tc.TransferFiles(ctx, req.login, req.serverID+":0", cfg)
 	if err != nil {
+		if errors.As(err, new(*sftp.NonRecursiveDirectoryTransferError)) {
+			return nil, trace.Errorf("transferring directories through the Web UI is not supported at the moment, please use tsh scp -r")
```

Comment:
Is there a better way to "rethrow" an error while keeping the stack trace? There's no `trace.Rewrap`.

I couldn't really use `trace.Wrap`, even if `lib/sshutils/sftp` returned a generic message, because the Web UI fundamentally handles those wrapped messages different than tsh. They're not shown before the wrapped error, they go into a separate field in the returned JSON object.

Reply:
There's no function in `gravitational/trace` for that, but if you really want you can do it with something like

```go
func rewrap(err, newErr error) error {
  if traceError := (trace.Error)(nil); errors.As(err, &traceError) {
    traceErr := traceError.Clone()
    traceErr.Err = trace.Unwrap(newErr)
    return traceErr
  }
  return trace.Wrap(newErr)
}